lib/auth/webauthn/login.go

```diff
@@ -104,3 +117,173 @@ func (f *LoginFlow) Begin(ctx context.Context, user string) (*CredentialAssertio
 
 	return assertion, nil
 }
+
+// Finish is the second and last step of the LoginFlow.
+// It returns the MFADevice used to solve the challenge. If login is successful,
+// Finish has the side effect of updating the counter and last used timestamp of
+// the returned device.
+func (f *LoginFlow) Finish(ctx context.Context, user string, resp *CredentialAssertionResponse) (*types.MFADevice, error) {
+	parsedResp, err := parseCredentialResponse(resp)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// If the appid extension is present, then we must set RPID = AppID and ensure
+	// the credential comes from an U2F device.
+	rpID := f.Webauthn.RPID
+	var usingAppID bool
+	// TODO(codingllama): Consider ignoring appid and basing the decision solely
+	//  in the device type. May be safer than assuming compliance?
+	// Do not read from parsedResp here, extensions don't carry over.
+	appidExt, ok := resp.Extensions["appid"]
+	if ok {
+		// Let's be as lenient as we can with the contents of "appid".
+		var err error
+		usingAppID, err = strconv.ParseBool(strings.ToLower(fmt.Sprint(appidExt)))
```

Comment:
Why the `fmt.Sprint`?

Reply:
`appidExt` is an `interface{}`. If the clients are well-behaved we should get back a boolean, but since this may come from Javascript I decided to play it safe and go with the Sprintf/ToLower/ParseBool combo.